### DISCUSSION TITLE
TensorboardLogger not displaying metrics for LogisticRegression

### DISCUSSION THREAD

**Discussion by Kaszanas** [Lightning-AI/pytorch-lightning#12762] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I am currently building a big dataset and infrastructure using PyTorch Lightning. Unfortunately there is no precise information in the documentation.

My code can be seen as:
```python
datamodule = DataModule(
        transform=economy_average_vs_outcome,
        download=False,
        batch_size=2,
        num_workers=4,
    )
# Preparing the data:
datamodule.prepare_data()
datamodule.setup()
logistic_regression = LogisticRegression(input_dim=2 * 39, num_classes=2)
logger = TensorBoardLogger("tb_logs", name="Logistic Regression")
trainer = pl.Trainer(
        logger=logger,
        accelerator="gpu",
        devices=1,
        auto_select_gpus=True,
        max_epochs=50,
        log_every_n_steps=2,
    )


# Training the model:
trainer.fit(model=logistic_regression, datamodule=datamodule)
```

Keep in mind that this is sample code that is not final.
Independent of that there is no Tensorboard visualizations available.

![image](https://user-images.githubusercontent.com/34846245/163412636-00dc7934-8cd6-49e6-8dc7-8a4dbca2257e.png)

![image](https://user-images.githubusercontent.com/34846245/163412749-c3f96aa6-d2ed-4b6e-b2d0-1fe41eaefad4.png)

The code for ```LogisticRegression``` is imported from ```pl_bolts``` as follows:

```
from pl_bolts.models.regression import LogisticRegression
```

I should not be required to read the source code for the ```LogisticRegression``` to figure out what is the internal logging implementation.

The ```training_step``` is as follows coming from ```pl_bolts```:
```python
    def training_step(self, batch: Tuple[Tensor, Tensor], batch_idx: int) -> Dict[str, Tensor]:
        x, y = batch

        # flatten any input
        x = x.view(x.size(0), -1)

        y_hat = self.linear(x)

        # PyTorch cross_entropy function combines log_softmax and nll_loss in single function
        loss = F.cross_entropy(y_hat, y, reduction="sum")

        # L1 regularizer
        if self.hparams.l1_strength > 0:
            l1_reg = self.linear.weight.abs().sum()
            loss += self.hparams.l1_strength * l1_reg

        # L2 regularizer
        if self.hparams.l2_strength > 0:
            l2_reg = self.linear.weight.pow(2).sum()
            loss += self.hparams.l2_strength * l2_reg

        loss /= x.size(0)

        tensorboard_logs = {"train_ce_loss": loss}
        progress_bar_metrics = tensorboard_logs
        return {"loss": loss, "log": tensorboard_logs, "progress_bar": progress_bar_metrics}
```

**Comment by rohitgr7** [comment, marked as answer]
the code in bolts is outdated and logging doesn't work like that anymore.
you need to do:
```py
def training_step(self, batch, batch_idx):
    ...
    self.log("train_ce_loss", loss, prog_bar=True)
```

more info herE: https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html#logging-from-a-lightningmodule

**Reply by Kaszanas** [reply]
I will try implementing that in my code and see if it solves the problem.

I might throw a PR into the bolts with this being fixed. It seems to be a relatively small fix that should be caught by some tests ;)
